src/npc.cpp

```diff
@@ -1655,7 +1655,7 @@ bool npc::wants_to_buy( const item &it, int at_price, int /*market_price*/ ) con
     }
 
     // TODO: Base on inventory
-    return at_price >= 80;
+    return at_price > 0;
```

Comment:
If market_price were actually passed in, this could be `(at_price >= 0) && (at_price <= market_price)`.

Reply:
This will often not work. `at_price` the price at which the merchant **wants** to buy. Those, how much he **needs** this item. And if he needs it, for example, if this food, and he is hungry, or if it is a weapon that is better than he has, this price will be higher than the base price. Let's not stop a merchant from buying things that he **really needs** (for a hight pice too). This is not socialism :)